ui/core-react/src/core-react/dialog/SimpleDialog.tsx

```diff
@@ -0,0 +1,245 @@
+  /** List of DialogButtonDef objects specifying buttons and associated onClick events */
+  buttonCluster?: DialogButtonDef[];
+
+  /** onClick event for X button for dialog */
+  onClose?: () => void;
+  /** 'keyup' event for Esc key */
+  onEscape?: () => void;
+  /** Triggered when a click is triggered outside of this dialog. */
+  onOutsideClick?: (event: MouseEvent) => any;
+
+  /** Whether to show background overlay. Default: true.
+   * @note Modeless dialogs require an id and an implementation of onModelessPointerDown.
+   */
+  modal?: boolean;
+  /** An id for a modeless dialog */
+  modelessId?: string;
+  /** Pointer Down event handler when modeless (modal = false) */
+  onModelessPointerDown?: (event: React.PointerEvent, id: string) => void;
+
+  /** Initial width of dialog. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. Default: "50%" */
+  width?: string | number;
+  /** Initial height of dialog. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. */
+  height?: string | number;
+  /** Minimum width that the dialog may be resized to. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. Default: 300px */
+  minWidth?: string | number;
+  /** Minimum height that the dialog may be resized to. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. Default: 100px */
+  minHeight?: string | number;
+  /** Maximum width that the dialog may be resized to. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. */
+  maxWidth?: string | number;
+  /** Maximum height that the dialog may be resized to. Displayed in px if value is a number; otherwise, displayed in specified CSS unit. */
+  maxHeight?: string | number;
+
+  /** Custom CSS Style for overlay */
+  backgroundStyle?: React.CSSProperties;
+  /** Custom CSS Style for title */
+  titleStyle?: React.CSSProperties;
+  /** Custom CSS Style for footer */
+  footerStyle?: React.CSSProperties;
+  /** Custom CSS class name for the content */
+  contentClassName?: string;
+  /** Custom CSS Style for the content */
+  contentStyle?: React.CSSProperties;
+}
+
+/** itwinui-react [[Dialog]] wrapper that excludes buggy features and adds features from core-react [[Dialog]].
+ * @internal
+ */
+export class SimpleDialog extends React.Component<SimpleDialogProps> {
+  private _parentDocument = document;
+  public static defaultProps: Partial<SimpleDialogProps> = {
+    minWidth: 300,
+    minHeight: 100,
+    width: "50%",
+    modal: true,
+    hideHeader: false,
+    trapFocus: false,
+  };
+
+  constructor(props: SimpleDialogProps) {
+    super(props);
+  }
+
+  public override componentWillUnmount(): void {
+    this._parentDocument.removeEventListener("keyup", this._handleKeyUp, true);
+  }
+
+  public override componentDidMount(): void {
+    this._parentDocument.addEventListener("keyup", this._handleKeyUp, true);
+  }
+
+  // istanbul ignore next
```

Comment:
This is a very temporary internal implementation until we move to itwinui-react `Dialog`, also the ignored lines are tested on core-react `Dialog` so in theory if something goes wrong there it should go wrong here as well. Knowing that, is it really necessary to duplicate test for this temporary solution?

Reply:
If that's the case, then it would be probably simpler to subclass the `Dialog` class, override the `render` and make the needed shared part protected. (Then the tests that tests the base class will not need to be duplicated and if a fix is made in the Dialog (even though unlikely) they will get caught in this implementation as well.) (and that would prevent you to rewrite the whole thing as a function too :) )

Reply:
Can you refactor the `getFooterButton` so we dont have this duplication given the size of the function, it could take optional parameters `primaryStyleType?` and `noCoreButtonClasses?`. The less we have to maintain later on, the better we'll be. (Also, please let me resolve my comments so I know if I'm happy with the resolution or not)